modules/c++/mt/include/mt/GenerationThreadPool.h

```diff
@@ -105,6 +107,35 @@ namespace mt
 	    waitGroup();
 	}
 
+    
+    /*!
+     *  \brief Runs a given operation on a sequence of numbers in parallel
+     *
```

Comment:
*personally* I'd prefer to see new code use C++11's task-based threading rather than adding to code in **mt**; but have at it ...

Reply:
Would that mean something like std::async? I definitely need more fine-grained control than that for the code I'm using this for.

Reply:
I definitely would rather use standard C++ threading utilities, but C++ doesn't have an official thread pool class so I had to use the CODA one.